controllers/keystoneapi_controller.go

```diff
@@ -266,6 +267,39 @@ func (r *KeystoneAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *KeystoneAPIReconciler) reconcileDelete(ctx context.Context, instance *keystonev1.KeystoneAPI, helper *helper.Helper) (ctrl.Result, error) {
 	l := GetLog(ctx)
 	l.Info("Reconciling Service delete")
+
+	// We need to allow all KeystoneEndpoint and KeystoneService processing to finish
+	// in the case of a delete before we remove the finalizers.  For instance, in the
+	// case of the Memcached dependency, if Memcached is deleted before all Keystone
+	// cleanup has finished, then the Keystone logic will likely hit a 500 error and
+	// thus its deletion will hang indefinitely.
+	for _, finalizer := range instance.Finalizers {
+		// If this finalizer is not our KeystoneAPI finalizer, then it is either
+		// a KeystoneService or KeystoneEndpointer finalizer, which indicates that
+		// there is more Keystone processing that needs to finish before we can
+		// allow our DB and Memcached dependencies to be potentially deleted
+		// themselves
+		if finalizer != helper.GetFinalizer() {
+			return ctrl.Result{}, nil
+		}
+	}
```

Comment:
should this be a simple lib-common func and a generic check in the all the operators being called right at the top of delete?